File: module/SampleModule/src/main/java/com/pds/sample/application/ModuleApplication.java
package com.pds.sample.application;

import android.app.Application;

import com.pds.base.manager.ActivityLifecycle;
import com.pds.router.ModuleRouter;
import com.pds.skin.SkinManager;
import com.pds.tool.ToolApplicationManager;
import com.pds.web.X5SDK;

/**
 * @author: pengdaosong
 * @CreateTime: 2020/9/26 2:31 PM
 * @Email: pengdaosong@medlinker.com
 * @Description:
 */
public class ModuleApplication {

    private static final String TAG = "MainApplication";
    private static ModuleApplication sModuleApplication = new ModuleApplication();
    private Application mApplication;

    private ModuleApplication(){}

    public static ModuleApplication instance(){
        return sModuleApplication;
    }

    public void onCreate(Application application) {
        mApplication = application;
        ModuleRouter.init(application);
        SkinManager.init(mApplication);
        X5SDK.init(application);
        ToolApplicationManager.onCreate(application);
        application.registerActivityLifecycleCallbacks(ActivityLifecycle.getInstance());
    }

    public Application application(){
        return mApplication;
    }
}

File: IM/socket_app/debug/java/blog.pds.com.socket.app/ModuleApplication.kt
package blog.pds.com.socket.app

import androidx.multidex.MultiDexApplication

/**
 * @author: pengdaosong
 * CreateTime:  2019-06-22 17:57
 * Email：pengdaosong@medlinker.com
 * Description:
 */
class ModuleApplication : MultiDexApplication(){

    companion object{
        private lateinit var application: ModuleApplication

        const val ip = "192.168.2.2"
        const val port = 6666

        fun app(): ModuleApplication {
            return application
        }
    }

    override fun onCreate() {
        super.onCreate()
        application = this
        SocketManager.init(this)
    }

}